enigma-swing/src/main/java/cuchaz/enigma/gui/GuiController.java

```diff
@@ -121,12 +123,26 @@ public void closeJar() {
 		this.gui.onCloseJar();
 	}
 
-	public CompletableFuture<Void> openMappings(MappingFormat format, Path path) {
-		if (this.project == null) return CompletableFuture.completedFuture(null);
+	public void openMappings(Path path) {
+		if (Files.isDirectory(path)) {
+			this.openMappings(MappingFormat.ENIGMA_DIRECTORY, path);
+		} else if ("zip".equalsIgnoreCase(MoreFiles.getFileExtension(path))) {
```

Comment:
It's probably a good idea to also support tiny / recaf / whatever other formats exist while you're at the whole moving-things-around thing

Reply:
I couldn't find a consistent file extension for any of the others, but I added tiny support!